tests/is-valid-utf8/Main.hs

```diff
@@ -0,0 +1,290 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Main (main) where
+
+import Data.Bits (shiftR, (.&.), shiftL)
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import Data.Char (chr, ord)
+import Data.Word (Word8)
+import Test.QuickCheck (Property, forAllShrink, (===))
+import Test.QuickCheck.Arbitrary (Arbitrary (arbitrary, shrink))
+import Test.QuickCheck.Gen (oneof, Gen, choose, vectorOf, listOf1, sized, resize,
+                            elements)
+import Test.Tasty (defaultMain, testGroup)
+import Test.Tasty.QuickCheck (testProperty)
+
+main :: IO ()
+main = defaultMain . testGroup "UTF-8 validation" $ [
+  testProperty "Valid UTF-8" goValid,
+  testProperty "Invalid UTF-8" goInvalid
+  ]
+  where
+    goValid :: Property
+    goValid = forAllShrink arbitrary shrink $
```

Comment:
Do we need `forAllShrink`? My understanding was that it does not bear much utility if you call it with stock `arbitrary` and `shrink`.

Reply:
This is mostly done out of habit. I do recall using just `forAll`, but _not_ getting shrinks before. This might not be an issue though, so I'll try replacing it.